graphium/data/sampler.py

```diff
@@ -0,0 +1,27 @@
+import torch.utils.data as data_utils
+import numpy as np
+
+
+class CustomSampler(data_utils.Sampler):
+    def __init__(self, dataset, sampler_task_dict):
+        self.dataset = dataset
+        self.sampler_task_dict = sampler_task_dict
+        self.task_indices = {}
+        for i in range(len(dataset)):
+            # the dataset[i]["labels"].keys() are not deterministic, need to sort by key length
```

Comment:
I suspect this for loop would take too long, even though it does only once at the beginning. Thinking of a way to cache it and reload.

Reply:
cache added, now use much less time with the `__init__` part.